crates/red_knot_python_semantic/src/types.rs

```diff
@@ -518,6 +529,17 @@ impl<'db> Type<'db> {
             {
                 true
             }
+            (Type::ClassLiteral(self_class), Type::Type(target_class)) => {
+                self_class.class.is_subclass_of(db, target_class.class)
+            }
+            (Type::Type(self_class), Type::Type(target_class)) => {
+                self_class.class.is_subclass_of(db, target_class.class)
+            }
+            (Type::Type(..), Type::Instance(InstanceType { class, .. }))
+                if class.is_known(db, KnownClass::Type) =>
```

Comment:
I think we can add a TODO comment here; once https://github.com/astral-sh/ruff/pull/14120 lands, we should flesh out this case (and the reverse case), both here and in `is_equivalent_to`, to fully establish the equivalence of `Type::Type(classA)` and `Type::Instance(classA.metaclass)`.